### PR TITLE
Keep snippet lists up to date & remove errors

### LINKED ISSUE #535: Clean up console warnings
## Expected Behavior

The console should be as free of React warnings as possible, and should have _no_ `TypeError` warnings.

## Current Behavior

Some React warnings are showing up that are related to the snippet list. The following error is also being logged:
```
TypeError: t.children is undefined
```

## Browser

This occurs in all browsers.

### LINKED ISSUE #536: Snippet lists can become stale
## Expected Behavior

Both the user's personal snippet list, as well as their organization lists should remain up to date at all times.

## Current Behavior

The user's snippet list is not updated when only an annotation is changed. The organization lists are not updated when a new snippet is created, or an existing one is modified.

## Browser

All the browsers.

### PR DESCRIPTION
**Pull request by Hopding**
## Description

This PR causes the user and organization snippet lists to be updated whenever a snippet is saved or updated (important to keep timestamps from becoming stale). Some console warnings & errors are also fixed.

## Motivation and Context

Fixes #535. Fixes #536.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Browsers

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Mobile:
- [ ] Other:

### CODE REVIEW

File: src/actions/app.js [hunk 1]
```diff
@@ -124,6 +125,9 @@ export const saveNew = org => (dispatch, getState) => {
         dispatch(addNotification('Codesplaination Saved!'));
         // Clear the hasUnsavedChanges flag
         dispatch(saveSucceeded());
+        // Update the user's personal and organization snippet lists
+        dispatch(updateUserSnippets());
+        dispatch(fetchSnippetLists());
```
Comment (solkaz) [inline src/actions/app.js]: `fetchSnippetLists` will fetch the snippets for the user and the organizations he/she belongs to, so the call to `updateUserSnippets` is unnecessary

File: src/actions/app.js [hunk 2]
```diff
@@ -166,6 +170,9 @@ export const saveExisting = () => (dispatch, getState) => {
...
```
Comment (solkaz) [inline src/actions/app.js]: See above

File: src/containers/SnippetArea.jsx [hunk 3]
```diff
@@ -230,7 +229,6 @@ export class SnippetArea extends React.Component {
           canEdit: true,
         }; // Grant all permissions, this is now her file.
         dispatch(setPermissions(permissions));
-        dispatch(updateUserSnippets());
       });
```
Comment (solkaz) [inline src/containers/SnippetArea.jsx]: What else are we able to refactor from this section? Anything that doesn't involve the state of the component or concerns the router could be moved to the then clauses of the thunks

File: src/components/Editor.jsx [hunk 4]
```diff
@@ -170,11 +170,16 @@ class Editor extends React.Component {
...
-    const { filters, AST } = this.props;
-    highlight(codeMirror, AST, filters);
+    // Always highlight when the cursor moves and text is in the editor
+    const { filters, AST, value } = this.props;
+
+    // Need to keep this type of check for the AST, otherwise, for a
+    // new snippet, we'll try to highlight with an AST === {}, which
+    // !== undefined
+    if (value && Object.keys(AST) > 0) {
+      const codeMirror = this.codeMirror.getCodeMirror();
```
Comment (dane-johnson) [inline src/components/Editor.jsx]: I'd move this to the top of the function with the other variable declarations

**Comment by thebho** [comment]
Is this ready?

**Pull request merged**